Спринт 6

RecoilSPB opened the pull request.

Recoilvrn committed:
[+] Добавлен метод remove(int id) в интерфейс HistoryManager
[!] Убрано ограничение для истории
[+] Реализован метод remove(int id) в InMemoryHistoryManager
[!] Изменен тип history с List на HashMap

Recoilvrn committed:
[!] Возращаемый тип в методе getHistory вернул на List

Recoilvrn committed:
[+] Добавил проверку на null в InMemoryHistoryManager.add
[!] Поправил тест checkHistoryLimit

Recoilvrn committed:
[+] Создал CustLinkedList
[!] Перевел InMemoryHistoryManager на CustLinkedList

Recoilvrn committed:
[-] Убрал не используемый импорт

Recoilvrn committed:
[!] Исправление CodeStyle

z8q commented on src/ru/practicum/service/InMemoryHistoryManager.java:
```diff
@@ -1,28 +1,26 @@
-    public InMemoryHistoryManager() {
-        this.history = new LinkedList<>();
-    }
+    CustLinkedList myLinkedList = new CustLinkedList();
```
Можно поставить private final

z8q commented on src/ru/practicum/utils/CustLinkedList.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    public void removeNode(int id) {
+        unlink(id);
```
тут получается как обертка, но никакого дополнительного функционала нет.
Возможно было бы лучше перенести сюда всю логику

z8q commented on src/ru/practicum/utils/Node.java:
```diff
@@ -0,0 +1,52 @@
+
+public class Node<E> {
+    private Node<E> prev;
+    private E e;
```
можно было бы назвать поле value. "e" похоже на exception)

Recoilvrn committed:
[!] CustLinkedList - Убрал метод unlink, логику перенес в removeNode
[!] InMemoryHistoryManager - Добавил модификаторы доступа private final
[!] Node - Переименовал поле "e" в value
[!] InHistoryManagerTest - Исправил имена вызываеммых методов
